src/instructlab/model/evaluate.py

```diff
@@ -301,6 +304,15 @@ def get_backend(backend, model):
     return backend
 
 
+def dump_results_to_json(benchmark: str, results: dict):
```

Comment:
The most common way to do this is to send the json results to stdout.  I understand that's an issue with ilab because because lots of other info is going to stdout.  But having a file with a timestamp in it makes it difficult to use this programatically.  Options to consider:

- Let the user specify the output file
- Move all the output during evaluation to a logfile, then return the json to stdout (this is my preference)

Reply:
Makes sense to me - for option 2 would writing all the evaluation output to a logfile happen just when we are outputting JSON, or all the time?

@cdoern we should consider a global `--silent` flag to disable all logger output for stuff like this

Reply:
> Makes sense to me - for option 2 would writing all the evaluation output to a logfile happen just when we are outputting JSON, or all the time?

Consistency across commands is probably most important.  We have a mixture today of:

1. Logging which can be broken down into
  1a. User friendly
  1b. Logging which is really debug
    1ba. Ideally these would either be moved to debug or made more friendly 
2. - Prints/echos intended for the user
3. - Final output from the command

The easiest thing to do is move 1 and 2 to a log.  That's probably the right thing from a REST API perspective as well (although the prints/echos should maybe be skipped/avoided if not run through the cli).

Note: The logging can already be sent to a log file if configured that way.

You could make an argument for leaving 1 and 2 in place if using "text" and sending them to a log if "json".  But it's unclear at the moment how we want users to interact with logs vs stdout.

Reply:
Revisiting this - I think I'll go with this approach you mentioned for now
```
You could make an argument for leaving 1 and 2 in place if using "text" and sending them to a log if "json". But it's unclear at the moment how we want users to interact with logs vs stdout.
```
How can I do the logging configuration to send to a file already that you mentioned?

Reply:
Was implemented here: https://github.com/instructlab/instructlab/pull/2084

Reply:
Pushing up something along these lines, still need to manually test it but it should be doing the following:
1. redirecting logging output to a {benchmark}_{timestamp}.log file in {output_dir} and writing the JSON results directly to stdout instead of a file